controllers/vaultconnection_controller.go

```diff
@@ -76,21 +79,46 @@ func (r *VaultConnectionReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 
 	var errs error
+	o.Status.Conditions = []metav1.Condition{}
 	vaultClient, err := vault.MakeVaultClient(ctx, vaultConfig, r.Client)
 	if err != nil {
 		logger.Error(err, "Failed to construct Vault client")
-		r.Recorder.Eventf(o, corev1.EventTypeWarning, consts.ReasonVaultClientError, "Failed to construct Vault client: %s", err)
+		r.Recorder.Eventf(o, corev1.EventTypeWarning,
+			consts.ReasonVaultClientError,
+			"Failed to construct Vault client: %s", err)
+
+		o.Status.Conditions = append(o.Status.Conditions,
+			newConditionNow(o, "VaultClient",
+				consts.ReasonVaultClientConfigError,
+				metav1.ConditionFalse,
+				"Failed to setup Vault client, address=%s, errs=%s",
+				o.Spec.Address, errs),
+		)
 
 		errs = errors.Join(errs, err)
 	}
 
 	if vaultClient != nil {
 		if _, err := vaultClient.Sys().SealStatusWithContext(ctx); err != nil {
 			logger.Error(err, "Failed to check Vault seal status, requeuing")
-			r.Recorder.Eventf(o, corev1.EventTypeWarning, "VaultClientError", "Failed to check Vault seal status: %s", err)
+			r.Recorder.Eventf(o, corev1.EventTypeWarning,
+				"VaultClientError",
+				"Failed to check Vault seal status: %s", err)
 			errs = errors.Join(errs, err)
+
+			o.Status.Conditions = append(o.Status.Conditions,
+				newConditionNow(o, "VaultPing",
```

Comment:
I see some of the other conditions const values have "Error" or "Failure" in them. Maybe we want this condition to be VaultPingError or VaultPingFailure? Otherwise it will might not seem like a problematic condition.